src/signed-data-store/signed-data-store.ts

```diff
@@ -0,0 +1,116 @@
+import { clearInterval } from 'timers';
+import { BigNumber, ethers } from 'ethers';
+import { LocalSignedData, SignedData, AirnodeAddress, DataStore, TemplateId } from '../types';
+import { logger } from '../logger';
+
+// A simple in-memory data store implementation - the interface allows for swapping in a remote key/value store
+
+const signedApiStore: Record<AirnodeAddress, Record<TemplateId, LocalSignedData>> = {};
+let pruner: NodeJS.Timeout | undefined;
+
+export const checkMessage = ({ airnode, templateId, timestamp, signature, encodedValue }: SignedData) => {
+  // 'data' is: ethers.utils.defaultAbiCoder.encode(['int256'], [beaconValue]);
+
+  const message = ethers.utils.arrayify(
+    ethers.utils.solidityKeccak256(['bytes32', 'uint256', 'bytes'], [templateId, timestamp, encodedValue])
+  );
+
+  const signerAddr = ethers.utils.verifyMessage(message, signature);
+  return signerAddr !== airnode;
+};
+
+export const checkSignedDataIntegrity = (signedData: SignedData) => {
+  const { airnode, templateId, timestamp, encodedValue } = signedData;
+
+  if (parseInt(timestamp) * 1_000 > Date.now()) {
+    logger.warn(
+      `Refusing to store sample as timestamp is in the future: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        parseInt(timestamp) * 1_000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  if (checkMessage(signedData)) {
+    logger.warn(
+      `Refusing to store sample as signature does not match: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        parseInt(timestamp) * 1_000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  return true;
+};
+
+const setStoreDataPoint = async (signedData: SignedData) => {
```

Comment:
This doesn't need to be `async`, right?
```suggestion
const setStoreDataPoint = (signedData: SignedData) => {
```

Reply:
No, but I figured, in line with other functions in this repo that we might want to make this async so we can drop in a key/value store. It might be early optimisation but async paths are infectious and changing it later on could be a serious pain.

Reply:
I should have probably commented why I did it that way :D